lighthouse-core/audits/dobetterweb/uses-request-compression.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * @fileoverview Audit a page to ensure that resources loaded with
+ * gzip/br/deflate compression.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const URL = require('../../lib/url-shim');
+const Formatter = require('../../formatters/formatter');
+
+const KB_IN_BYTES = 1024;
+const TOTAL_WASTED_BYTES_THRESHOLD = 100 * KB_IN_BYTES;
+
+class ResponsesAreCompressed extends Audit {
```

Comment:
It's been a while and this might be a pain (sorry!), but there's a new `byte-efficiency` folder and base audit you can extend that should eliminate a bit of duplicated code